packages/bff/src/services/authorizationService.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  AuthToken,
+  CustomClaims,
+  InteropTokenGenerator,
+  Logger,
+  ORGANIZATION,
+  ORGANIZATION_EXTERNAL_ID_CLAIM,
+  ORGANIZATION_EXTERNAL_ID_ORIGIN_CLAIM,
+  ORGANIZATION_EXTERNAL_ID_VALUE_CLAIM,
+  ORGANIZATION_ID_CLAIM,
+  SELFCARE_ID_CLAIM,
+  SessionClaims,
+  SessionTokenGenerator,
+  USER_ROLES,
+  decodeJwtToken,
+  verifyJwtToken,
+} from "pagopa-interop-commons";
+import { genericError } from "pagopa-interop-models";
+import {
+  invalidJwtClaim,
+  missingClaim,
+  unknownTenantOrigin,
+} from "../model/domain/errors.js";
+import { PagoPAInteropBeClients } from "../providers/clientProvider.js";
+import { config } from "../config/config.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function authorizationServiceBuilder(
+  interopTokenGenerator: InteropTokenGenerator,
+  sessionTokenGenerator: SessionTokenGenerator,
+  tenantProcessClient: PagoPAInteropBeClients["tenantProcessClient"],
+  allowList: string[]
+) {
+  const readJwt = async (
+    identityToken: string,
+    logger: Logger
+  ): Promise<{
+    roles: string;
+    sessionClaims: SessionClaims;
+    selfcareId: string;
+  }> => {
+    const verified = await verifyJwtToken(identityToken, logger);
+    if (!verified) {
+      throw invalidJwtClaim("Token verification failed");
```

Comment:
in Scala this error is thrown when specifically checking claims.
If I understand, `verifyJwtToken` validates also the signature, so it could be worth to change the error name and its message